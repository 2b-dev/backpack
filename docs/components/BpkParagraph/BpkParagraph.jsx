import React, { PropTypes } from 'react'
import CssModules from 'react-css-modules'

import styles from './bpk-paragraph.scss'

const BpkParagraph = ({ children }) => (
  <p styleName='bpk-paragraph'>{children}</p>
)

BpkParagraph.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}

export default CssModules(BpkParagraph, styles)
