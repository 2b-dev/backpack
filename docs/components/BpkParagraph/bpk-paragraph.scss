@import './../../../scss/bonds';
@import './../../../scss/mixins/typography';

.bpk-paragraph {
  margin-top: $bpk-margin-top-p;
  margin-bottom: $bpk-margin-bottom-p;
}
