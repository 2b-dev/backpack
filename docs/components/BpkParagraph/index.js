import component from './BpkParagraph.jsx'

export default component
