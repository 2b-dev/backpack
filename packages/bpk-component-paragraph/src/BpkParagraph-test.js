import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';
import BpkParagraph from './BpkParagraph';

describe('BpkParagraph', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<BpkParagraph>My paragraph</BpkParagraph>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
