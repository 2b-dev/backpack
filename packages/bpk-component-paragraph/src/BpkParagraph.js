import React, { PropTypes } from 'react'

import './bpk-paragraph.scss'

const BpkParagraph = (props) => <p className='bpk-paragraph' children={props.children} />

BpkParagraph.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default BpkParagraph

