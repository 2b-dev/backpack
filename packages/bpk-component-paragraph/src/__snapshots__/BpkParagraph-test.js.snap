exports[`BpkParagraph should render correctly 1`] = `
<p
  className="bpk-paragraph">
  My paragraph
</p>
`;
